resolver/resolver.cc

```diff
@@ -1788,11 +1788,29 @@ class ResolveSanityCheckWalk {
 
 vector<ast::ParsedFile> Resolver::run(core::MutableContext ctx, vector<ast::ParsedFile> trees, WorkerPool &workers) {
     trees = ResolveConstantsWalk::resolveConstants(ctx, std::move(trees), workers);
+    if (ctx.state.shouldCancelTypechecking()) {
+        return {};
```

Comment:
this makes method terminate "as-if" it succeeded.
Let's either change the result time to indicate cancellation or use exceptions